<?xml version="1.0"?>

<launch>
  
  <arg name="camera"   default="stereo"/>





    <node pkg="nodelet" type="nodelet" name="points_xyzrgb" args="standalone rtabmap_ros/point_cloud_xyzrgb" respawn="true" output="screen">

        <remap from="left/image"        to="$(arg camera)/left/image_rect"/>
        <remap from="right/image"       to="$(arg camera)/right/image_rect"/>
        <remap from="left/camera_info"  to="$(arg camera)/left/camera_info"/>
        <remap from="right/camera_info" to="$(arg camera)/right/camera_info"/>

        <remap from="cloud"             to="$(arg camera)/voxel_cloud" />
        <remap from="depth_ghc"         to="$(arg camera)/depth_ghc/image" />

        <param name="decimation"  type="double" value="1"/>
        <param name="voxel_size"  type="double" value="0.02"/>
        <param name="approx_sync" type="bool"   value="false"/>

        <param name="isMilliMeter" type="bool"  value="false"/>

        <param name="max_depth" type="double" value="5.0"/>
        <param name="min_depth" type="double" value="0.0"/>

        <param name="offset_t" type="double" value="20"/>  <!-- depthimage crop: top  //0 is invalid -->
        <param name="offset_b" type="double" value="100"/>  <!-- depthimage crop: bottom //0 is invalid -->

        <param name="frame_id_depth" value="$(arg camera)_depth_optical_frame"/>
        <param name="frame_id_cloud" value="$(arg camera)_depth_optical_frame"/>

        <!-- <param name="Grid/NoiseFilteringRadius" type="double" value="0.5"/> -->
    </node>


</launch>
